pkg/distributor/receiver/middleware_test.go

```diff
@@ -0,0 +1,77 @@
+package receiver
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/weaveworks/common/user"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/model/pdata"
+	"google.golang.org/grpc/metadata"
+)
+
+type assertFunc func(*testing.T, context.Context)
+
+type testConsumer struct {
+	t          *testing.T
+	assertFunc assertFunc
+}
+
+func (tc *testConsumer) Capabilities() consumer.Capabilities {
```

Comment:
Just checking: you've seen the testing facilities, right? Not sure you need them though.

https://github.com/open-telemetry/opentelemetry-collector/blob/839909863acceff104c18a9db1b5d24c13d6a972/consumer/consumertest/sink.go#L121-L128

Reply:
need Capabilities
![image](https://user-images.githubusercontent.com/9583245/154189947-913b51c8-04bd-44d3-9d09-278588f5db2a.png)

Reply:
If you use the helper to create a consumer, the helper will implement this function for you.